src/main/java/frc/robot/Apriltags.java

```diff
@@ -0,0 +1,164 @@
+    public static final double kLensHeight = 23.5; // Camera lens height 
+    public static final double kMountAngle = 0; // Camera lens angle
+    public static final double maxArmReach = 10; //Maximum reach in inches the arm can stretch outside frame
+    public static final double minArmReach = 5; //Minimum reach in inches the arm can strecth outside frame
+    public static final double bumperThickness = 3;
+       private NetworkTable table;
+       Drive drive = Robot.drive;
+       
+           public Apriltags()
+           {
+               table = NetworkTableInstance.getDefault().getTable("limelight");
+           }
+
+           public Apriltags(String name) {
+               table = NetworkTableInstance.getDefault().getTable(name);
+           }
+       
+           public Apriltags(NetworkTable table) {
+               this.table = table;
+           }
+
+   
+           /**
+            * @return whether or not the limelight has any valid targets
+            */
+           public boolean isTargetFound() {
+
+               return table.getEntry("tv").getDouble(0) == 0f; 
+               // changed this to "not" as it was returning opposite
+           }
+       
+           /**
+            * @return the horizontal offset from crosshair to target (-29.8 to +29.8 degrees)
+            */
+           public double getHorizontalDegToTarget() {
+            double horizontalDegOffset = table.getEntry("tx").getDouble(0);
+
+            return horizontalDegOffset;
+           }
+       
+           /**
+            * @return vertical offset from crosshair to target (-24.85 to +24.85 degrees)
+            */
+           public double getVerticalDegToTarget() {
+            double verticalDegOffset = table.getEntry("ty").getDouble(0);
+            
+            return verticalDegOffset;
+           }
+       
+           /**
+            * @param pipeline sets the limelight’s current pipeline
+            */
+           public void setPipeline(int pipeline) {
+               if (pipeline < 0) {
+                   pipeline = 1;
+                   throw new IllegalArgumentException("Pipeline can not be less than zero");
+               } else if(pipeline > 9) {
+                   pipeline = 2;
+                   throw new IllegalArgumentException("Pipeline can not be greater than one");
+               }
+               table.getEntry("pipeline").setValue(pipeline);
+               table.getEntry("pipeline").setNumber(pipeline);
+           }
+       
+           /**
+            * @return current pipeline of limelight
+            */
+           public double getPipeline() {
+               return table.getEntry("pipeline").getDouble(0.0);
+           }
+       
+           /**
+            * @return distance from target in inches
+           */
+           public double estimateHorizontalDistance(){
+               double veritcalTargetAngle = getVerticalDegToTarget();
+               double height = kTargetHeight - kLensHeight;
```

Comment:
I'd recommend using absolute value here, to account for whether the camera height is above or below the target just in case

```suggestion
               double height = Math.abs(kTargetHeight - kLensHeight);
```